### PR TITLE
Adds update channel bookmark sort order endpoint

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR adds the update channel bookmarks sort order endpoint. Sits on top of https://github.com/mattermost/mattermost/pull/25653, so it should be merged after it

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54420

#### Release Note
```release-note
Added update channel bookmark sort order endpoint at /api/v4/channels/{channel_id}/bookmarks/{bookmark_id}/sort_order.
```

### CODE REVIEW

File: server/channels/api4/channel_bookmark.go [hunk 1]
```diff
@@ -196,3 +197,95 @@ func updateChannelBookmark(c *Context, w http.ResponseWriter, r *http.Request) {
...
+
+	connectionID := r.Header.Get(model.ConnectionId)
+
+	c.RequireChannelId()
+	if c.Err != nil {
+		return
+	}
+
+	var newSortOrder int64
+	if err := json.NewDecoder(r.Body).Decode(&newSortOrder); err != nil {
```
Comment (enahum) [inline server/channels/api4/channel_bookmark.go:215]: Are you expecting just a number in the body? No key/value or something similar? Can you expand on the reason?

File: server/channels/api4/channel_bookmark_test.go
Comment (enahum) [inline server/channels/api4/channel_bookmark_test.go:1]: Missing trying to set the new order value to something irrational like negative or above the number of existing bookmarks

**Review by enahum — CHANGES_REQUESTED** [review]

File: server/channels/api4/channel_bookmark_test.go
Comment (mgdelacroix) [reply server/channels/api4/channel_bookmark_test.go:1]: Added on https://github.com/mattermost/mattermost/pull/25686/commits/27071f17dd71dc8f34ded787cbbfa4d08c35c6b7, PTAL

File: server/channels/api4/channel_bookmark.go (hunk 1 above)
Comment (mgdelacroix) [reply server/channels/api4/channel_bookmark.go:215]: The endpoint only requires one piece of information, and due its format that's unlikely to change. I'm 1/5 on this tho, so if we feel strongly about having a JSON object, I'm happy to update the PR

Comment (enahum) [reply server/channels/api4/channel_bookmark.go:215]: No strong opinions just wanted to know

**Review by enahum — APPROVED** [review]

**Pull request merged**